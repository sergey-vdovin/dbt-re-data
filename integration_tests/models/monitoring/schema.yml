version: 2

models:
  - name: test_re_data_metrics
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_metrics')

  - name: test_re_data_z_score
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_z_score')

  - name: test_re_data_anomalies
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_anomalies')

  - name: test_re_data_test_history
    tests:
     - dbt_utils.equality:
         compare_model: ref('expected_test_history')