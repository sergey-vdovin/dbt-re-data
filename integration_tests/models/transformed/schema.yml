version: 2

models:
  - name: buy_events
    tests:
      - re_data.assert_in_range:
          metric: row_count
          min_value: 0
          max_value: 10

      - re_data.assert_equal:
          metric: row_count
          value: 1

      - re_data.assert_false:
          metric: freshness
          expression: value is null
      
      - re_data.assert_greater_equal:
          metric: my_distinct_table_rows
          value: 10

    columns:
      - name: value1
        tests:
          - re_data.assert_in_range:
              metric: nulls_percent
              min_value: 0
              max_value: 10

          - re_data.assert_true:
              metric: nulls_percent
              expression: value = 0
          
          - re_data.assert_less:
              metric: min
              value: 100
              where: time_window_start >= '2020-05-02'
          
          - re_data.assert_less_equal:
              metric: min
              value: 107

      - name: value2
        tests:
          - re_data.assert_greater:
              metric: min
              value: 200
              where: time_window_start >= '2020-05-02'
