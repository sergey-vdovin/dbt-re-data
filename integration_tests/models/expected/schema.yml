version: 2

models:
  - name: test_re_data_row_count
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_row_count')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - metric
            - value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_freshness
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_freshness')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - metric
            - value
            - time_window_end

  - name: test_re_data_max
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_max')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_max_length
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_max_length')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_z_score
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_z_score')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - cast (z_score_value * 1000 as integer) as z_score_value
            - cast (modified_z_score_value * 1000 as integer) as modified_z_score_value
            - time_window_end
            - cast (last_value * 1000 as integer) as last_value
            - cast (last_avg * 1000 as integer) as last_avg
            - cast (last_stddev * 1000 as integer) as last_stddev
            - cast (last_median * 1000 as integer) as last_median
            - cast (last_iqr * 1000 as integer) as last_iqr
            - cast (last_median_absolute_deviation * 1000 as integer) as last_median_absolute_deviation
            - cast (last_mean_absolute_deviation * 1000 as integer) as last_mean_absolute_deviation
            - interval_length_sec

  - name: test_re_data_anomalies
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_anomalies')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - anomaly_detector
            - interval_length_sec

  - name: test_re_data_metrics
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_metrics')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - cast (value * 1000 as integer) as value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_test_history
    tests:
     - dbt_utils.equality:
         compare_model: ref('expected_sample_data_test_history')
         compare_columns:
          - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
          - coalesce(upper(column_name), '') as column_name
          - right(test_name, 15) as test_name
          - status